redis/lookup.go

```diff
@@ -0,0 +1,154 @@
+package redis
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/letsencrypt/boulder/cmd"
+	blog "github.com/letsencrypt/boulder/log"
+
+	"github.com/redis/go-redis/v9"
+)
+
+// Lookup is a helper that keeps *redis.Ring shards up to date using SRV
+// lookups.
+type Lookup struct {
+	// service is the symbolic name of the desired service.
+	service string
+
+	// domain is the domain name of the desired service.
+	domain string
+
+	// updateFrequency is the frequency of periodic SRV lookups. Defaults to 30
+	// seconds.
+	updateFrequency time.Duration
+
+	// dnsAuthority is the single <hostname|IPv4|[IPv6]>:<port> of the DNS
+	// server to be used for SRV lookups. If the address contains a hostname it
+	// will be resolved via the system DNS. If the port is left unspecified it
+	// will default to '53'. If this field is left unspecified the system DNS
+	// will be used for resolution.
```

Comment:
This section seems redundant because it's already mentioned that "if the address contains a hostname it will be resolved via system DNS."
> If this field is left unspecified the system DNS
>	// will be used for resolution.

Or is it that if there is no `dnsAuthority` configured, that the `cmd.ServiceDomain` will be resolved by system DNS?

Reply:
It's the latter 👍🏻